groups/ntc/ntci/ntci_streamsocket.h

```diff
@@ -1502,6 +1506,16 @@ class StreamSocket : public ntsi::Descriptor,
     virtual ntsa::Error setReadQueueWatermarks(bsl::size_t lowWatermark,
                                                bsl::size_t highWatermark) = 0;
 
+    /// Request the implementation to start timestamping outgoing data if the
+    /// specified 'enable' flag is true. Otherwise, request the implementation
+    /// to stop timestamping outgoing data. Return the error.
+    virtual ntsa::Error timestampOutgoingData(bool enable);
```

Comment:
Let's remember that if someone decides to enable tx timestamps for a stream socket dynamically then we may fail to correlate timestamps later.

Reply:
Would you prefer if we fail the function if the socket is already connected?

Reply:
I have implemented your suggested option 2) in the comment about `ntcr::StreamSocket::timestampOutgoingData()`, below.